planning/obstacle_velocity_planner/src/node.cpp

```diff
@@ -0,0 +1,689 @@
+// Copyright 2022 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "obstacle_velocity_planner/node.hpp"
+
+#include "obstacle_velocity_planner/polygon_utils.hpp"
+#include "obstacle_velocity_planner/utils.hpp"
+#include "tier4_autoware_utils/ros/update_param.hpp"
+#include "tier4_autoware_utils/trajectory/tmp_conversion.hpp"
+
+#include <boost/format.hpp>
+
+#include <algorithm>
+#include <chrono>
+
+namespace
+{
+VelocityLimitClearCommand createVelocityLimitClearCommandMsg(const rclcpp::Time & current_time)
+{
+  VelocityLimitClearCommand msg;
+  msg.stamp = current_time;
+  msg.sender = "obstacle_velocity_planner";
+  msg.command = true;
+  return msg;
+}
+
+Trajectory trimTrajectoryFrom(const Trajectory & input, const geometry_msgs::msg::Point & position)
```

Comment:
Node: It is better to consider the posture(yaw) when searching for the nearest point.

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/570/commits/1b0f40b2e0703e04f651dd87250358f88d03af3d

Note: We don't use max dist/yaw to calculate nearest index of obstacle to trajectory